### PR TITLE
fix all compilation errors for OpenCL support

### PR DESCRIPTION
**Pull request by edgarriba**

### CODE REVIEW

File: src/device.cpp [hunk 1]
```diff
@@ -71,20 +84,27 @@ int device::workgroup_size(int id) {
...
   if (backend_ == BACKEND_CUDA) {
 #ifdef USE_CUDA
     return 1;
 #endif  // USE_CUDA
   } else {
 #ifdef USE_OPENCL
-    return OPENCL_QUEUE_COUNT;
+    // TODO(naibaf7): where do you set this value?
+    // return OPENCL_QUEUE_COUNT;
+    return 1;
```
Comment (edgarriba) [inline src/device.cpp:102]: @naibaf7 Could you check what value should have `OPENCL_QUEUE_COUNT`?

**Comment by edgarriba** [comment]
@naibaf7 What format do you want to keep? There's a mix of CamelCase and underscore_score formats.

**Comment by naibaf7** [comment]
@edgarriba 
Not quite sure. Need to re-check with Intel, as they have added in some of that stuff, and I'm not sure with what style we should be going. According to google style guide it is OK to have CamelCase for heavy functions and under_score format for accessors/getters (Regular functions have mixed case; "cheap" functions may use lower case with underscores. ref https://google.github.io/styleguide/cppguide.html).
Caffe has been done that way and I think I'll keep the mix.

OPENCL_QUEUE_COUNT is deprecated for the moment, any value is fine.

**Comment by edgarriba** [comment]
@naibaf7 nice !

File: include/common.hpp [hunk 2]
```diff
@@ -5,7 +5,9 @@
 #include <type_traits>
 #include <sstream>
 #include <iomanip>
+
 #include "version.hpp"
+#include "greentea_libdnn_config.h"
```
Comment (edgarriba) [inline include/common.hpp:10]: @naibaf7 When I install greentea this file it's not found. What do you suggest to do with this file during installation?

**Pull request merged**